stl/inc/atomic

```diff
@@ -480,13 +490,37 @@ struct _Atomic_storage<_Ty, 1> { // lock-free using 1-byte intrinsics
 
     bool compare_exchange_strong(_Ty& _Expected, const _Ty _Desired,
         const memory_order _Order = memory_order_seq_cst) noexcept { // CAS with given memory order
-        const char _Expected_bytes = _Atomic_reinterpret_as<char>(_Expected); // read before atomic operation
+        char _Expected_bytes = _Atomic_reinterpret_as<char>(_Expected); // read before atomic operation
         char _Prev_bytes;
-        _ATOMIC_CHOOSE_INTRINSIC(_Order, _Prev_bytes, _InterlockedCompareExchange8, _Atomic_address_as<char>(_Storage),
-            _Atomic_reinterpret_as<char>(_Desired), _Expected_bytes);
-        if (_Prev_bytes == _Expected_bytes) {
-            return true;
+
+#if _HAS_CXX20
+        if constexpr (_STD has_unique_object_representations_v<_Ty>) {
+#endif
+            _ATOMIC_CHOOSE_INTRINSIC(_Order, _Prev_bytes, _InterlockedCompareExchange8,
+                _Atomic_address_as<char>(_Storage), _Atomic_reinterpret_as<char>(_Desired), _Expected_bytes);
+            if (_Prev_bytes == _Expected_bytes) {
+                return true;
+            }
+#if _HAS_CXX20
+        } else {
+            _Ty _Mask;
+            _CSTD memset(&_Mask, 0xff, sizeof(_Ty));
+            __builtin_zero_non_value_bits(_STD addressof(_Mask));
+            const short _Mask_val = _Atomic_reinterpret_as<short>(_Mask);

```

Comment:
Is `short` copy-pasted? This is `_Atomic_storage<_Ty, 1>` and everything else here says `char` and `8`.